app/Models/Bible/BibleBook.php

```diff
@@ -119,4 +138,51 @@ public function book()
     {
         return $this->belongsTo(Book::class);
     }
+
+    /**
+     * Get collection of bible books sorted by bible versification or the book_seq column if it not empty.
+     *
+     * @param string $bible_id
+     * @param string $book_id
+     * @param string $bible_versification
+     *
+     * @return Collection
+     */
+    public static function getAllSortedByBookSeqOrVersification(
+        string $bible_id,
+        ?string $book_id = null,
+        ?string $bible_versification = null
+    ) : Collection {
+        $select_columns = [
+            'bible_books.bible_id',
+            'bible_books.book_id',
+            'bible_books.name',
+            'bible_books.name_short',
+            'bible_books.chapters',
+            'bible_books.book_seq'
+        ];
+
+        if ($bible_versification) {
+            $select_columns[] = \DB::raw(
+                'CASE
+                    WHEN bible_books.book_seq IS NOT NULL THEN bible_books.book_seq
+                    WHEN books.'.$bible_versification.'_order IS NOT NULL THEN books.'.$bible_versification.'_order
+                    ELSE bible_books.book_id
+                END AS book_by_order'
+            );
+        }
+
+        return BibleBook::select($select_columns)
+            ->where('bible_id', $bible_id)
+            ->when($book_id, function ($query) use ($book_id) {
+                $query->where('book_id', $book_id);
+            })
+            ->join('books', 'books.id', 'bible_books.book_id')
+            ->with('book')
+            ->when($bible_versification, function ($order_by_query) {
```

Comment:
change this to just an order-by clause, without the "when". It will make more sense

Reply:
@bradflood , the above line has been replaced by:

```php
->orderBy('book_by_order')
```